SITES-22935 NGDM remote image issue

mohiaror opened the pull request:
Make changes work for multiple fileupload instances present in a custom multifield component.

<!--
Before making a PR please make sure to read our contributing guidelines
https://github.com/adobe/aem-core-wcm-components/blob/master/CONTRIBUTING.md

IMPORTANT: Please base your pull request on the **main** branch and make sure to check you have incorporated or merged the latest changes!

For issue references: Add a comma-separated list of a [closing word](https://help.github.com/articles/closing-issues-via-commit-messages/)
followed by the ticket number fixed by the PR. It should be underlined in the preview if done correctly.
-->

| Q                        | A <!--(Can use an emoji 👍) -->
| ------------------------ | ---
| Fixed Issues?            | `Fixes #1, Fixes #2` <!-- remove the (`) quotes to link the issues -->
| Patch: Bug Fix?          |
| Minor: New Feature?      |
| Major: Breaking Change?  |
| Tests Added + Pass?      | Yes
| Documentation Provided   | Yes (code comments and or markdown)
| Any Dependency Changes?  |
| License                  | Apache License, Version 2.0

<!-- Describe your changes below in as much detail as possible -->

mohiaror committed:
SITES-22935 NGDM remote image issue

Make changes work for multiple fileupload instances present in a custom multifield component.

mohiaror commented:
@YahorC  I could not understand the change done in this commit - https://github.com/adobe/aem-core-wcm-components/commit/bfc2bd6ce1d7363158d4bec01eb78f90aa3bc372

I am not sure if that change is related to remoteReference (which are used for polaris' connected assets usecase). I tried to incorporate that fix in my PR. Can you please check if it is Ok?

cbogdan0707 commented on content/src/content/jcr_root/apps/core/wcm/components/image/v3/image/clientlibs/editor/js/image.js:
```diff
@@ -108,35 +108,27 @@
+                        var $fileReferences = $cqFileUpload.find("[data-cq-fileupload-parameter='filereference']");
+                        $fileReferences.each(function() {
+                            var remoteFileReference = $(this).val();
+                            if (remoteFileReference && remoteFileReference !== "" &&
```
if(remoteFileReference && remoteFilreReference !== ""),  seems to me redundant since an empty string is falsy

cbogdan0707 commented on content/src/content/jcr_root/apps/core/wcm/components/image/v3/image/clientlibs/editor/js/image.js:
```diff
@@ -108,35 +108,27 @@
-                            if (areDMFeaturesEnabled && !isPolarisEnabled) {
-                                selectPresetType($(presetTypeSelector), "imagePreset");
-                                resetSelectField($dynamicMediaGroup.find(smartCropRenditionDropDownSelector));
+                    if (fileReference === undefined) {
```
do you need to explicitly check if the fileReference is undefined? having to check if it is falsy, you could simplify this to be (!fileReference)

mohiaror committed:
SITES-22935 NGDM remote image issue

Incorporate review comments.

LSantha commented:
@mohiaror , could you please add UI test for this fix?

mohiaror committed:
Merge branch 'main' into issues/SITES-22935

YahorC commented:
@mohiaror Changes that I applied was regarding page properties image section
So for Image component everything was working fine, but page properties sharing the same logic, but node structure, to which we are referring while trying to fetch image path are different for page properties Image section and just image component
Generally fix is looking fine for me, but could you test please, if it works fine for page properties image tab, to make sure it is working fine https://jira.corp.adobe.com/browse/SITES-21939

mohiaror committed:
Merge branch 'main' into issues/SITES-22935

mohiaror committed:
SITES-22935 NGDM remote image issue

Incorporate changes done in other PR - https://github.com/adobe/aem-core-wcm-components/commit/b5e6a560f3d4c29a68313ae07b9670b265067d7a

vladbailescu committed:
Merge branch 'main' into issues/SITES-22935

vladbailescu committed:
Merge branch 'main' into issues/SITES-22935

RaduADumitru committed:
Merge branch 'main' into issues/SITES-22935

vladbailescu commented:
This is being fixed in the main product code
